doc/example/patterns_invoke_functor.cpp

```diff
@@ -38,7 +46,8 @@ namespace
 BOOST_AUTO_TEST_CASE( how_to_invoke_a_functor_passed_as_parameter_of_a_mock_method )
 {
     mock_class mock;
-    MOCK_EXPECT( mock.method ).calls( boost::bind( boost::apply< void >(), _1, 42 ) ); // whenever 'method' is called, invoke the functor with 42
+    MOCK_EXPECT( mock.method ).calls( [](const auto &functor){ functor(42); } ); // whenever 'method' is called, invoke the functor with 42
     function( mock );
+    BOOST_CHECK(receivedValue == 42);
```

Comment:
This shows up in the docs but the definition of `receivedValue` doesn’t.

Reply:
I wanted to keep the exposed code minimal. How shall we do this? Add a comment above this that as an example `function` calls `c.method` with a functor that sets "receivedValue" to the argument passed?

Reply:
Couldn’t we move the check out of the test and into a test fixture destructor? Or maybe it’s too late there for failures to happen?

Reply:
The test fixture destructor can't assert (terminate would be called), upgrade to Boost 1.59 or 1.65 is required. (see https://github.com/mat007/turtle/pull/88#discussion_r494589023)

I added a comment

Reply:
Isn’t there a way to mark the test as failed without throwing an exception?

Or 1.65, that’s fine as well 😁